src/main/scala/com/quartethealth/spark/fixedwidth/FixedwidthRelation.scala

```diff
@@ -0,0 +1,56 @@
+package com.quartethealth.spark.fixedwidth
+
+import com.databricks.spark.csv.CsvRelation
+import com.databricks.spark.csv.readers.{BulkReader, LineReader}
+import com.quartethealth.spark.fixedwidth.readers.{BulkFixedwidthReader, LineFixedwidthReader}
+import org.apache.spark.rdd.RDD
+import org.apache.spark.sql.SQLContext
+import org.apache.spark.sql.types.StructType
+
+class FixedwidthRelation protected[spark] (
+    baseRDD: () => RDD[String],
+    fixedWidths: Array[Int],
+    location: Option[String],
+    useHeader: Boolean,
+    parseMode: String,
+    comment: Character,
+    ignoreLeadingWhiteSpace: Boolean,
+    ignoreTrailingWhiteSpace: Boolean,
+    treatEmptyValuesAsNulls: Boolean,
+    userSchema: StructType,
+    inferSchema: Boolean,
+    codec: String = null,
+    nullValue: String = "")(@transient override val sqlContext: SQLContext)
+  extends CsvRelation(
```

Comment:
You're right, it doesn't make a lot of sense from an ideal design perspective. However I wanted to make as few changes as possible to the original source, so this was a compromise. I'd be willing to be persuaded otherwise though...